seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/catalog/PaimonCatalogLoader.java

```diff
@@ -51,24 +50,35 @@ public class PaimonCatalogLoader implements Serializable {
     private static final String HDFS_IMPL_KEY = "fs.hdfs.impl";
 
     private String warehouse;
+    private PaimonCatalogEnum catalogType;
+    private String catalogUri;
 
     private PaimonHadoopConfiguration paimonHadoopConfiguration;
 
-    public PaimonCatalogLoader(PaimonSinkConfig paimonSinkConfig) {
-        this.warehouse = paimonSinkConfig.getWarehouse();
-        this.paimonHadoopConfiguration = PaimonSecurityContext.loadHadoopConfig(paimonSinkConfig);
+    public PaimonCatalogLoader(PaimonConfig paimonConfig) {
+        this.warehouse = paimonConfig.getWarehouse();
+        this.catalogType = paimonConfig.getCatalogType();
+        this.catalogUri = paimonConfig.getCatalogUri();
+        this.paimonHadoopConfiguration = PaimonSecurityContext.loadHadoopConfig(paimonConfig);
     }
 
     public Catalog loadCatalog() {
         // When using the seatunel engine, set the current class loader to prevent loading failures
         Thread.currentThread().setContextClassLoader(PaimonCatalogLoader.class.getClassLoader());
         final Map<String, String> optionsMap = new HashMap<>(1);
-        optionsMap.put(WAREHOUSE.key(), warehouse);
-        final Options options = Options.fromMap(optionsMap);
+        optionsMap.put(CatalogOptions.WAREHOUSE.key(), warehouse);
         if (warehouse.startsWith(HDFS_PREFIX)) {
             checkConfiguration(paimonHadoopConfiguration, HDFS_DEF_FS_NAME);
             paimonHadoopConfiguration.set(HDFS_IMPL_KEY, HDFS_IMPL);
         }
+        if (PaimonCatalogEnum.HIVE.getType().equals(catalogType.getType())) {
+            optionsMap.put(CatalogOptions.METASTORE.key(), PaimonCatalogEnum.HIVE.getType());
```

Comment:
Would it be better if we moved it outside of 'if' in accordance with the original catalog type, even if it defaults to hadoop when not set?

Reply:
> Would it be better if we moved it outside of 'if' in accordance with the original catalog type, even if it defaults to hadoop when not set?

The default is the hadoop catalog, which does not need to be specified separately.

Reply:
Yes, I'm just worried that this feature will change with this paimon release

Reply:
> Yes, I'm just worried that this feature will change with this paimon release

The code here only works when catalog is hive and does not affect the logic when catalog is hadoop.

Reply:
ok,I'm sorry. Maybe I worry too much